server/lib/validation/languages/th.js

```diff
@@ -56,43 +56,43 @@ const INVALIDATIONS = [{
   // - Tone marks
   // - Phinthu, Thanthakhat, Nikhahit, Yamakkan
   /* eslint-disable-next-line no-misleading-character-class */
-  regex: /(^|\s+)[\u0E30\u0E32\u0E33\u0E45\u0E31\u0E34\u0E35\u0E36\u0E37\u0E4D\u0E47\u0E38\u0E39\u0E48\u0E49\u0E4A\u0E4B\u0E3A\u0E4C\u0E4D\u0E4E]/,
+  regex: /(^|\s)[\u0E30\u0E32\u0E33\u0E45\u0E31\u0E34\u0E35\u0E36\u0E37\u0E4D\u0E47\u0E38\u0E39\u0E48\u0E49\u0E4A\u0E4B\u0E3A\u0E4C\u0E4D\u0E4E]/,
   error: 'Word should not start with unexpected characters',
 }, {
   // These Thai chars cannot end the word:
   // - Lead vowels
-  regex: /[\u0E40\u0E41\u0E42\u0E43\u0E44](\s+|$)/,
+  regex: /[\u0E40\u0E41\u0E42\u0E43\u0E44](\s|$)/,
   error: 'Word should not end with leading vowels',
 }, {
   // Any words consisting of letters with a period
   // inbetween are considered abbreviations or acronyms.
   // Abbreviations in Latin chars are disallowed by previous rules already.
-  regex: /[ก-ฮ]+\.([ก-ฮ]+\.)+/,
+  regex: /[ก-ฮ]\.[ก-ฮ]{1,5}\./,
   error: 'Sentence should not contain abbreviations',
 }, {
-  // Five or more repeating consonants in a row is likely a non-formal spelling or difficult to read.
-  regex: /[ก-ฮ]{5,}/,
-  error: 'Sentence should not contain 5 or more consonants in a row',
+  // Seven or more repeating characters in a row is likely a non-formal spelling or difficult to read.
+  regex: /(.)\1{6}/,
```

Comment:
Is that change based on new data?

Reply:
- The previous regex will catch any 5+ characters in the same class.
- That is buggy, as the intention is to catch only 5+ characters of the same character.
- It should catch only: `กกกกก` and not `กฮกกฮ`.
  - So using, `( )\1` to refer to the first group that got caught.

- The number of repeating characters has changed from 5 to 7, just to be more conservative.

- `[ก-ฮ]` class of chars has also changed to `.`, to generalized the rule and reject any repeating of 7+ chars (indicator of bad quality text)